src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/junit/JUnit5Strategy.kt

```diff
@@ -68,7 +72,9 @@ object JUnit5Strategy : AbstractTestStrategy() {
                 """
                             public void executionFinished(org.junit.platform.engine.TestDescriptor testDescriptor, org.junit.platform.engine.TestExecutionResult testExecutionResult) {
                                 mainRunner.executionFinished(testDescriptor, testExecutionResult);
-                                ${TestListener::class.java.name}.INSTANCE.${TestListener::testFinished.name}(testDescriptor.getUniqueId().toString(), testExecutionResult.getStatus().name());
+                                if(testDescriptor.getType() == org.junit.platform.engine.TestDescriptor.Type.TEST) {
+                                    ${TestListener::class.java.name}.INSTANCE.${TestListener::testFinished.name}(testDescriptor.getUniqueId().toString(), testExecutionResult.getStatus().name());
```

Comment:
testDescriptor.getUniqueId().toString() duplicates three times - maybe in const or variable

Reply:
I don't think it's possible to put this in a constant